Code/GraphMol/FileParsers/MaeMolSupplier.cpp

```diff
@@ -473,7 +495,69 @@ void MaeMolSupplier::moveToNextBlock() {
   } catch (const mae::read_exception &e) {
     d_stored_exc = e.what();
   }
+  ++d_position;
+}
+
+bool MaeMolSupplier::atEnd() {
+  if (d_next_struct == nullptr) {
+    d_length = d_position;
+    return true;
+  }
+  return false;
+}
+
+unsigned int MaeMolSupplier::length() {
+  PRECONDITION(dp_inStream, "no stream");
+
+  if (d_length == 0 && !atEnd()) {
+    // maeparser has an internal buffer, so we can't just iterate over
+    // block till we reach the end of the file. So we have to rewind
+    // the input stream, use it to create a separate parser, fast
+    // forward this one to the end of the data, and then get the length
+    // from that parser. Then we can restore the input stream to
+    // the position where it was before, so that it is still in
+    // sync with maeparser's internal buffer.
+
+    dp_sInStream->clear();
+    auto current_position = dp_sInStream->tellg();
+    dp_sInStream->seekg(0, std::ios::beg);
+
+    MaeMolSupplier tmp_supplier(dp_sInStream);
```

Comment:
Question: is there no faster (or at least more direct) way to do this using the underlying `mae:Reader` object?

Reply:
Sadly, I believe there isn't. It's something we should add, but at the moment I think this is the best we can do.